pkg/agent/run.go

```diff
@@ -40,91 +43,98 @@ var Flags AgentCmdFlags
 const schemaVersion string = "v2.0.0"
 
 // Run starts the agent process
-func Run(cmd *cobra.Command, args []string) {
-	logs.Log.Printf("Preflight agent version: %s (%s)", version.PreflightVersion, version.Commit)
-	ctx, cancel := context.WithCancel(context.Background())
+func Run(cmd *cobra.Command, args []string) (runErr error) {
+	log := Flags.LogOptions.Initialize()
+
+	ctx, cancel := context.WithCancel(klog.NewContext(context.Background(), log))
 	defer cancel()
 
+	log.Info("starting", "version", version.PreflightVersion, "commit", version.Commit)
+
 	file, err := os.Open(Flags.ConfigFilePath)
 	if err != nil {
-		logs.Log.Fatalf("Failed to load config file for agent from: %s", Flags.ConfigFilePath)
+		return fmt.Errorf("Failed to load config file for agent from %q: %s", Flags.ConfigFilePath, err)
 	}
 	defer file.Close()
 
 	b, err := io.ReadAll(file)
 	if err != nil {
-		logs.Log.Fatalf("Failed to read config file: %s", err)
+		return fmt.Errorf("Failed to read config file: %s", err)
 	}
 
 	cfg, err := ParseConfig(b)
 	if err != nil {
-		logs.Log.Fatalf("Failed to parse config file: %s", err)
+		return fmt.Errorf("Failed to parse config file: %s", err)
 	}
 
-	config, preflightClient, err := ValidateAndCombineConfig(logs.Log, cfg, Flags)
+	config, preflightClient, err := ValidateAndCombineConfig(log, cfg, Flags)
 	if err != nil {
-		logs.Log.Fatalf("While evaluating configuration: %v", err)
+		return fmt.Errorf("Failed to evaluate configuration: %s", err)
 	}
 
+	serverMux := http.NewServeMux()
+
 	if Flags.Profiling {
-		logs.Log.Printf("pprof profiling was enabled.\nRunning profiling on port :6060")
-		go func() {
-			err := http.ListenAndServe(":6060", nil)
-			if err != nil && !errors.Is(err, http.ErrServerClosed) {
-				logs.Log.Fatalf("failed to run pprof profiler: %s", err)
-			}
-		}()
```

Comment:
To remove this particular instance of `log.Fatalf` I need to replace these unmanaged go  routines with a `errgroup.Go` so that I can catch the errors and return them.